examples/src/main/scala/za/co/absa/fadb/examples/enceladus/DatasetSchema.scala

```diff
@@ -60,72 +59,63 @@ object DatasetSchema {
 
-  private implicit val SchemaHeaderImplicit: GetResult[SchemaHeader] = GetResult(r => {SchemaHeader(r.<<, r.<<)})
-  private implicit val GetSchemaImplicit: GetResult[Schema] = GetResult(r => {
-    val status: Int = r.<<
-    val statusText: String = r.<<
-    if (status != 200) {
-      throw new StatusException(status, statusText)
-    }
-    Schema(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
-  })
-
-  final class AddSchema(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBUniqueFunction[ExecutorEngineType, SchemaInput, Long](schema)
-    with SlickPgFunction[SchemaInput, Long] {
+  final class AddSchema(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBUniqueFunction[SchemaInput, Long, SlickPgEngine]
+    with SlickPgFunctionWithStatusSupport[SchemaInput, Long]
+    with UserDefinedStatusHandling {
 
-
-    override protected def sqlToCallFunction(values: SchemaInput): SQLActionBuilder = {
+    override protected def sql(values: SchemaInput): SQLActionBuilder = {
       sql"""SELECT A.status, A.status_text, A.id_schema_version
-             FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
-                ${values.fields}::JSONB, ${values.userName}
-             ) A;"""
+            FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
+              ${values.fields}::JSONB, ${values.userName}
+            ) A;"""
     }
 
-    override protected def resultConverter: GetResult[Long] = {
-      val gr:GetResult[Long] = GetResult(r => {
-        val status: Int = r.<<
-        val statusText: String = r.<<
-        if (status != 201) throw new StatusException(status, statusText)
-        r.<<
-      })
-      gr
-    }
+    override protected def slickConverter: GetResult[Long] = GetResult.GetLong
+
+    override def OKStatuses: Set[Integer] = Set(201)
+
   }
 
-  final class GetSchema(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBUniqueFunction[ExecutorEngineType, (String, Option[Int]), Schema](schema)
-    with SlickPgFunction[(String, Option[Int]), Schema] {
+  final class GetSchema(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBUniqueFunction[(String, Option[Int]), Schema, SlickPgEngine]
+    with SlickPgFunctionWithStatusSupport[(String, Option[Int]), Schema]
+    with UserDefinedStatusHandling {
 
     def apply(id: Long): Future[Schema] = {
       val sql =
         sql"""SELECT A.*
              FROM #$functionName($id) A;"""
 
-      schema.unique(makeQueryFunction(sql)(resultConverter))
+      val slickQuery = new dBEngine.QueryType(sql, slickConverter)
+      dBEngine.unique[Schema](slickQuery)
     }
 
-    override protected def sqlToCallFunction(values: (String, Option[Int])): SQLActionBuilder = {
+    override protected def sql(values: (String, Option[Int])): SQLActionBuilder = {
```

Comment:
What's this Option[Int] for? If it's schema name an version, then according to the `Schema` case class, they are both mandatory:

```
schemaName: String,
schemaVersion: Int,
```

Reply:
I thought it was in some description.
Version is mandatory with the schema (any entity), but when not provided, the latest version is returned.
Then it's perfectly OK, to use `None` = `Null` = _unknown_/_undefined_ in the query, me thinks.

Reply:
ooh okay. Yeah perhaps some quick code doc can be useful for these examples then. But I know that we'll rework it anyway..